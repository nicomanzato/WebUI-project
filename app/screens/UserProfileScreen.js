'use strict';

import React, {Component} from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  Text,
  Image,
  TouchableHighlight,
  ActivityIndicator
} from 'react-native';

import {connect} from 'react-redux';
import {requestUserProfile} from '../store/user/userActions'
import UserProfile from './../components/user/userProfile'
import PostList from './../components/post/postList'

export class UserProfileScreen extends Component {

  constructor(props) {
    super(props);

    this.state = {
      userId: -1,
    }
  }

  componentDidMount = () => {
    const userId = this.props.navigation.getParam('userId', 'NO-ID');
    this.setState({
      userId: userId,
    });

    this.props.requestUserProfile(userId);
  }

  render = () => {
    return (
     <View style={styles.container}>
       {(this.props.isLoadingUserProfile || this.props.isLoadingUserProfilePost) &&
         <View style={styles.activityIndicatorContainer}>
           <ActivityIndicator animating={true}/>
         </View>
       }
       {!this.props.isLoadingUserProfile && !this.props.isLoadingUserProfilePost &&
         <View style={styles.userContainer}>
           <UserProfile user={this.props.user} styles={styles.userProfile} />
           <View style={styles.separator} />
           <PostList
             style={styles.postList}
             navigation={this.props.navigation}
             data={this.props.userProfilePost}
             refreshing={false}
             onRefresh={() => {}}
             onEndReached={() => {}}
           />
        </View>
       }

     </View>
    );
  }

}

function mapStateToProps(state, props) {
    return {
      isLoadingUserProfile: state.UserReducer.isLoadingUserProfile,
      user: state.UserReducer.user,

      userProfilePost: state.PostReducer.userProfilePost,
      isLoadingUserProfilePost: state.PostReducer.isLoadingUserProfilePost,
    }
}

const mapDispatchToProps = {
  requestUserProfile: (userId) => requestUserProfile(userId),
}

export default connect(mapStateToProps, mapDispatchToProps)(UserProfileScreen);

const styles = StyleSheet.create({
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
  userProfile: {
    marginBottom: 5,
  },
  postList: {
    marginTop: 5,
  },
  userContainer: {
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'space-around',
    flexDirection: 'column',
  },
  container: {
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'space-around',
    flexDirection: 'column',
  },
  activityIndicatorContainer:{
    backgroundColor: "#fff",
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
});
