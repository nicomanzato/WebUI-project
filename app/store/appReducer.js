'use strict';

import { combineReducers } from 'redux'
import PostReducer from './post/postReducer'
import TrendReducer from './trend/trendReducer'
import UserReducer from './user/userReducer'
import nav from './navigation/navReducer'
import ConfigurationReducer from './configuration/configurationReducer'

const AppReducer = combineReducers({
  PostReducer,
  TrendReducer,
  UserReducer,
  nav,
  ConfigurationReducer,
});

export default AppReducer;
