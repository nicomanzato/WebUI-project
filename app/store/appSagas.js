import { call, all, put, takeEvery, takeLatest, take, select } from 'redux-saga/effects'
import { delay } from 'redux-saga'

import postSagas from './post/postSagas'
import trendSagas from './trend/trendSagas'
import userSagas from './user/userSagas'

export default function* appSagas() {
  yield all([
    postSagas(),
    trendSagas(),
    userSagas(),
  ])
}
