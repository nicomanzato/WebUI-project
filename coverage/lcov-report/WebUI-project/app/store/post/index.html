<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/store/post</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> WebUI-project/app/store/post
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.02% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>173/188</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.56% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>40/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>42/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.8% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>168/183</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="postActions.js"><a href="postActions.js.html">postActions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	</tr>

<tr>
	<td class="file high" data-value="postReducer.js"><a href="postReducer.js.html">postReducer.js</a></td>
	<td data-value="95.24" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="21" class="abs high">20/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	</tr>

<tr>
	<td class="file medium" data-value="postSagas.js"><a href="postSagas.js.html">postSagas.js</a></td>
	<td data-value="79.03" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 79%;"></div><div class="cover-empty" style="width:21%;"></div></div></td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="16" class="abs high">15/16</td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	</tr>

<tr>
	<td class="file high" data-value="postSelector.js"><a href="postSelector.js.html">postSelector.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 27 2018 17:02:42 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
