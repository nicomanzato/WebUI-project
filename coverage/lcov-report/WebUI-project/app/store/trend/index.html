<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/store/trend</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> WebUI-project/app/store/trend
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>26/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>26/30</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="trendActions.js"><a href="trendActions.js.html">trendActions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="trendReducer.js"><a href="trendReducer.js.html">trendReducer.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="trendSagas.js"><a href="trendSagas.js.html">trendSagas.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 26 2018 13:28:33 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
