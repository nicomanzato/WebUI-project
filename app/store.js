import { applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import appReducer from './store/appReducer'; //Import the reducer
import appSagas from './store/appSagas';
import reactotron from '../ReactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export const store = reactotron.createStore(appReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(appSagas)
