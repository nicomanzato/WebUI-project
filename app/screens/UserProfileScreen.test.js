import React from 'react';
import {UserProfileScreen} from './UserProfileScreen'
import ReduxWrappedUserProfileScreen from './UserProfileScreen'

import UserProfileMock from './../store/user/mock/UserProfileMock'
import PostMock from './../store/post/mock/postMock'

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import {store} from './../store'
import { shallow, mount, ReactWrapper } from 'enzyme';


describe('user profile screen', () => {

  it('should render correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}, getParam: () => { return '1'}}
    const wrapper = shallow(
      <UserProfileScreen
        navigation={mockNavigationProp}
        requestUserProfile={() => {}}
        isLoadingUserProfile={true}
        user={UserProfileMock}
        userProfilePost={{}}
        isLoadingUserProfilePost={false}
      />
     );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render search results correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}, getParam: () => { return '1'}}
    const wrapper = shallow(
      <ReduxWrappedUserProfileScreen
        store={store}
        navigation={mockNavigationProp}
        requestUserProfile={() => {}}
        isLoadingUserProfile={false}
        user={UserProfileMock}
        userProfilePost={[PostMock, PostMock, PostMock]}
        isLoadingUserProfilePost={false}
      />
     );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle dispatching its actions', () => {
    const requestUserProfile = jest.fn().mockName('mockedFunction');
    const mockNavigationProp = {state: {params: {data: '1'}}, getParam: () => { return '1'}}
    const wrapper = mount(
      <UserProfileScreen
        navigation={mockNavigationProp}
        requestUserProfile={requestUserProfile}
        isLoadingUserProfile={false}
        user={UserProfileMock}
        userProfilePost={[PostMock, PostMock, PostMock]}
        isLoadingUserProfilePost={false}
      />
     );
    expect(requestUserProfile).toHaveBeenCalled();
  });

});
