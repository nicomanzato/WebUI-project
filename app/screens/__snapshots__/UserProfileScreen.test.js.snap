// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user profile screen should render correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </Component>
</Component>
`;

exports[`user profile screen should render search results correctly 1`] = `
<UserProfileScreen
  isLoadingUserProfile={false}
  isLoadingUserProfilePost={false}
  navigation={
    Object {
      "getParam": [Function],
      "state": Object {
        "params": Object {
          "data": "1",
        },
      },
    }
  }
  requestUserProfile={[Function]}
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={Object {}}
  userProfilePost={Array []}
/>
`;
